intro/README.md

```diff
@@ -0,0 +1,561 @@
+Let's take a simple example:
+
+```c
+#include <stdio.h>
+
+void main()
+{
+    int a = 5;
+    int b[10];
+    b[11] = 9;
+    printf("%d\n", a);
+}
+```
+
+Compiling this code (without the stack protector activated) will lead to a buffer overflow in which the variable `a` will be modified and the print will output `9`.
+Compiling the same code with D (simply replace the include with an `import std.stdio : printf`) will issue a compile time error that states that 11 is beyond the size of the array.
+Aggregating the pointer of the array with its size facilitates a safer implementation of arrays.
+
+D implements 2 types of arrays:
+1. [Static Arrays](https://dlang.org/spec/arrays.html#static-arrays): their length must be known at compile time and therefore cannot be modified; all examples up until this point have been using static arrays.
+2. [Dynamic Arrays](https://dlang.org/spec/arrays.html#dynamic-arrays): their length may change dynamically at run time.
+
+### Slicing
+
+Slicing an array means to specify a subarray of it.
+An array slice does not copy the data, it is only another reference to it:
+
+```d
+void main()
+{
+    int[10] a;       // declare array of 10 ints
+    int[] b;
+ 
+    b = a[1..3];     // a[1..3] is a 2 element array consisting of a[1] and a[2]
+    int x = b[1];    // equivalent to `int x = 0;`
+    a[2] = 3;
+    int y = b[1];    // equivalent to `int y = 3;`
+}
+```
+
+### Array operations
+
+1. Array setting:
+
+    ```d
+    void main()
+    {
+        int[3] a = [1, 2, 3];
+        int[3] c = [ 1, 2, 3, 4];  // error: mismatched sizes
+        int[] b = [1, 2, 3, 4, 5];
+        a = 3;                     // set all elements of a to 3 
+        b = 5;                     // set all elements of b to 5
+        a[] = 2;                   // same as `a = 3`; using an empty slice is the same as slicing the full array
+        b = a[0 .. $];             // $ evaluates to the length of the array (in this case 10)
+        b = a[];                   // semantically equivalent to the one above
+        b = a[0 .. a.length]       // semantically equivalent to the one above
+        b[] = a[];                 // semantically equivalent to the one above
+        b[2 .. 4] = 4              // same as b[2] = 4, b[3] = 4
+        b[0 .. 4] = a[0 .. 4]      // error, a does not have 4 elements
+        a[0 .. 3] = b              // error, operands have different length
+    }
+    ```
+        
+    **.length** is a builtin array property.
+    For an extensive list of array properties click [here](https://dlang.org/spec/arrays.html#array-properties).
+
+    Now let's see how arrays work with functions.
+    The next quiz can be tricky but it will help us better understand how arrays work.
+    If it help, think what would happed in a similar situation in C.
+    
+    [Quiz](./quiz/ref-vs-value.md)
```

Comment:
This quiz requires knowledge about structs. Up until this point we haven't explained what structs are.

Reply:
I suggest putting this quiz after we talk about structs and after we explain the fact that an array in D is seen as a struct that contains the pointer and a size.